dso/compliancedb/model.py

```diff
@@ -0,0 +1,61 @@
+import datetime
+from enum import Enum
+
+import sqlalchemy
+from sqlalchemy import Column, ForeignKey
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship
+
+import dso.model
+
+
+Base = declarative_base()
+
+
+class ScanTool(Enum):
+    WHITESOURCE = 'whitesource'
+    PROTECODE = 'protecode'
+    CHECKMARX = 'checkmarx'
+    CLAMAV = 'clamav'
+
+
+class Artifact(Base):
+    __tablename__ = 'artifact'
+
+    id = Column(sqlalchemy.String, primary_key=True)
+    component_name = Column(sqlalchemy.String)
+    component_version = Column(sqlalchemy.String)
+    name = Column(sqlalchemy.String)
+    type = Column(sqlalchemy.types.Enum(
+        dso.model.ArtifactType,
+        values_callable=lambda v: [e.value for e in v],
+    ))
+    version = Column(sqlalchemy.String)
+
+    scan_result = relationship('ScanResult')
+
+    def __repr__(self):
+        return f'Scan(id={self.id!r} '\
+            f'Component(name={self.component_name!r}, version={self.component_version!r}), '\
+            f'Artifact(name={self.name!r}, type={self.type!r}, version={self.version!r}))'
+
+
+class ScanResult(Base):
+    __tablename__ = 'scan_result'
+
+    id = Column(sqlalchemy.String, primary_key=True)
+    artifact_id = Column(sqlalchemy.String, ForeignKey('artifact.id'))
+    timestamp = Column(
+        sqlalchemy.TIMESTAMP(timezone=True),
+        default=datetime.datetime.utcnow(),
+    )
+    source = Column(sqlalchemy.types.Enum(
+        ScanTool,
+    ))
+    scan_data = Column(sqlalchemy.JSON)
+
+    def __repr__(self):
+        return f'ScanResult(id={self.id!r}, '\
```

Comment:
is it actually necessary to define custom __repr__s? :-/
I thought sqalchemy would offer this (maybe through explicit cfg, which by default is not enabled)